### ISSUE TITLE
DISABLED test_torchscript_function (__main__.JitRpcTestWithSpawn)

### ISSUE THREAD

**Issue by mrshenli** [pytorch/pytorch#34304]
https://app.circleci.com/jobs/github/pytorch/pytorch/4702075

```
Mar 05 05:33:36 ...[E request_callback.cpp:39] Received error while processing request type 0: attempted to get undefined function python_function (get_function at /var/lib/jenkins/workspace/torch/csrc/jit/api/compilation_unit.h:69)
Mar 05 05:33:36 frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0x6c (0x7fefb74b257c in /opt/conda/lib/python3.6/site-packages/torch/lib/libc10.so)
Mar 05 05:33:36 frame #1: <unknown function> + 0x65977d (0x7fefc670c77d in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 05 05:33:36 frame #2: torch::distributed::rpc::RequestCallbackImpl::processRpc(torch::distributed::rpc::RpcCommandBase&, torch::distributed::rpc::MessageType, long) const + 0x151 (0x7fefc69d2961 in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 05 05:33:36 frame #3: torch::distributed::rpc::RequestCallbackImpl::processMessage(torch::distributed::rpc::Message&) const + 0x55 (0x7fefc69d50c5 in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 05 05:33:36 frame #4: torch::distributed::rpc::RequestCallback::operator()(torch::distributed::rpc::Message&) const + 0x2f (0x7fefc374467f in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_cpu.so)
Mar 05 05:33:36 frame #5: <unknown function> + 0x903cb8 (0x7fefc69b6cb8 in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 05 05:33:36 frame #6: c10::ThreadPool::main_loop(unsigned long) + 0x2b3 (0x7fefb74ac0f3 in /opt/conda/lib/python3.6/site-packages/torch/lib/libc10.so)
Mar 05 05:33:36 frame #7: <unknown function> + 0xc819d (0x7fefcafc019d in /opt/conda/lib/libstdc++.so.6)
Mar 05 05:33:36 frame #8: <unknown function> + 0x76ba (0x7feffd55e6ba in /lib/x86_64-linux-gnu/libpthread.so.0)
Mar 05 05:33:36 frame #9: clone + 0x6d (0x7feffd29441d in /lib/x86_64-linux-gnu/libc.so.6)
Mar 05 05:33:36 
Mar 05 05:33:37 .[E request_callback.cpp:39] Received error while processing request type 0: attempted to get undefined function nonexisting_script (get_function at /var/lib/jenkins/workspace/torch/csrc/jit/api/compilation_unit.h:69)
Mar 05 05:33:37 frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0x6c (0x7f448569757c in /opt/conda/lib/python3.6/site-packages/torch/lib/libc10.so)
Mar 05 05:33:37 frame #1: <unknown function> + 0x65977d (0x7f44948f177d in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 05 05:33:37 frame #2: torch::distributed::rpc::RequestCallbackImpl::processRpc(torch::distributed::rpc::RpcCommandBase&, torch::distributed::rpc::MessageType, long) const + 0x151 (0x7f4494bb7961 in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 05 05:33:37 frame #3: torch::distributed::rpc::RequestCallbackImpl::processMessage(torch::distributed::rpc::Message&) const + 0x55 (0x7f4494bba0c5 in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 05 05:33:37 frame #4: torch::distributed::rpc::RequestCallback::operator()(torch::distributed::rpc::Message&) const + 0x2f (0x7f449192967f in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_cpu.so)
Mar 05 05:33:37 frame #5: <unknown function> + 0x903cb8 (0x7f4494b9bcb8 in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 05 05:33:37 frame #6: c10::ThreadPool::main_loop(unsigned long) + 0x2b3 (0x7f44856910f3 in /opt/conda/lib/python3.6/site-packages/torch/lib/libc10.so)
Mar 05 05:33:37 frame #7: <unknown function> + 0xc819d (0x7f44991a519d in /opt/conda/lib/libstdc++.so.6)
Mar 05 05:33:37 frame #8: <unknown function> + 0x76ba (0x7f44cb7436ba in /lib/x86_64-linux-gnu/libpthread.so.0)
Mar 05 05:33:37 frame #9: clone + 0x6d (0x7f44cb47941d in /lib/x86_64-linux-gnu/libc.so.6)
Mar 05 05:33:37 
Mar 05 05:33:38 .[E request_callback.cpp:39] Received error while processing request type 0: The following operation failed in the TorchScript interpreter.
Mar 05 05:33:38 Traceback of TorchScript (most recent call last):
Mar 05 05:33:38   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/jit/rpc_test.py", line 140, in raise_script
Mar 05 05:33:38 @torch.jit.script
Mar 05 05:33:38 def raise_script():
Mar 05 05:33:38     raise RuntimeError("Expected error")
Mar 05 05:33:38     ~~~~~~~~~~~~~~~~~~~~ <--- HERE
Mar 05 05:33:38     return 0
Mar 05 05:33:38 RuntimeError: Exception
Mar 05 05:33:38 
Mar 05 05:34:47 .............Process process 2:
Mar 05 05:34:47 Traceback (most recent call last):
Mar 05 05:34:47   File "/opt/conda/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
Mar 05 05:34:47     self.run()
Mar 05 05:34:47   File "/opt/conda/lib/python3.6/multiprocessing/process.py", line 93, in run
Mar 05 05:34:47     self._target(*self._args, **self._kwargs)
Mar 05 05:34:47   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 243, in _run
Mar 05 05:34:47     getattr(self, test_name)()
Mar 05 05:34:47   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/dist_utils.py", line 68, in new_test_method
Mar 05 05:34:47     return_value = old_test_method(self, *arg, **kwargs)
Mar 05 05:34:47   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/jit/rpc_test.py", line 461, in test_torchscript_function
Mar 05 05:34:47     ret = rpc.rpc_sync(dst_worker_name, one_arg, args=(torch.ones(2, 2),))
Mar 05 05:34:47   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 76, in wrapper
Mar 05 05:34:47     return func(*args, **kwargs)
Mar 05 05:34:47   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 526, in rpc_sync
Mar 05 05:34:47     return fut.wait()
Mar 05 05:34:47 RuntimeError: RPC ran for more than 60000 milliseconds and timed out.
Mar 05 05:38:48 E.s.
Mar 05 05:38:48 ======================================================================
Mar 05 05:38:48 ERROR [300.214s]: test_torchscript_function (__main__.JitRpcTestWithSpawn)
Mar 05 05:38:48 ----------------------------------------------------------------------
Mar 05 05:38:48 Traceback (most recent call last):
Mar 05 05:38:48   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 175, in wrapper
Mar 05 05:38:48     self._join_processes(fn)
Mar 05 05:38:48   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 256, in _join_processes
Mar 05 05:38:48     self._check_return_codes(elapsed_time)
Mar 05 05:38:48   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 275, in _check_return_codes
Mar 05 05:38:48     raise RuntimeError('Process {} terminated or timed out after {} seconds'.format(i, elapsed_time))
Mar 05 05:38:48 RuntimeError: Process 0 terminated or timed out after 300.20289039611816 seconds
```

cc @ezyang @gchanan @zou3519 @suo @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @xush6528 @jjlilley @osalpekar

**Comment by zhaojuanmao** [comment]
GIL change in #33329 could cause deadlock

**Comment by mrshenli** [comment]
closed by #34505
